### DISCUSSION TITLE
This seems difficult to achieve. How can I display the current row number in the first column of a DataGrid and have it update accordingly when a new row is inserted or deleted?

### DISCUSSION THREAD

**Discussion by IronEaterBeast** [AvaloniaUI/Avalonia#15895] [Q&A] [answered]
For example, if there are rows 1/2/3 and I delete the second row, the first column of the third row should now display 2.

**Comment by maxkatz6** [comment, marked as answer]
Handle DataGrid.LoadingRow event, you will be able to edit row header content from there.
